crates/solver/src/solver/score_computation.rs

```diff
@@ -0,0 +1,180 @@
+// Suppose a solver generates a solution, and we compute the three variables:
+// surplus, fees and gas_cost. Currently, the objective is then simply equal to
+// surplus + fees - gas_cost. However, for CIP-20, we now have one additional
+// step for each solver. So for each solver we have trained a revert risk model,
+// which essentially is a function that computes the probability of the
+// settlement not reverting, and is a function of 3 parameters.
+// p (num_orders_in_sol, gas_used, gas_price).
+// So we compute this p, and then the score is equal to:
+// score = p * (surplus + fees) - gas_cost
+
+use {
+    super::SolverType,
+    crate::objective_value,
+    anyhow::{Context, Result},
+    clap::{Parser, ValueEnum},
+    ethcontract::U256,
+    num::ToPrimitive,
+    std::{
+        collections::HashMap,
+        fmt::{Display, Formatter},
+        ops::Neg,
+        str::FromStr,
+    },
+};
+
+#[derive(Debug, Default, Clone)]
+pub struct ScoreCalculator {
+    pub gas_amount_factor: f64,
+    pub gas_price_factor: f64,
+    pub nmb_orders_factor: f64,
+    pub intercept: f64,
+}
+
+impl ScoreCalculator {
+    pub fn calculate(&self, inputs: &objective_value::Inputs, nmb_orders: usize) -> Option<U256> {
+        let surplus = inputs.surplus_given.to_f64()?;
+        let fees = inputs.solver_fees.to_f64()?;
+        let gas_amount = inputs.gas_amount.to_f64()?;
+        let gas_price = inputs.gas_price.to_f64()?;
+        let exponent = self.intercept.neg()
+            - self.gas_amount_factor * gas_amount
+            - self.gas_price_factor * gas_price
+            - self.nmb_orders_factor * nmb_orders as f64;
```

Comment:
the scaling of `gas_amount` and `gas_price` is missing. otherwise the computation looks good.

(not sure if the following is valid rust)
```suggestion
        let exponent = self.intercept.neg()
            - self.gas_amount_factor * gas_amount * 1e-6
            - self.gas_price_factor * gas_price * 1e-1
            - self.nmb_orders_factor * nmb_orders as f64;
```

Reply:
Thanks, added.